Modules/database_manager.py

```diff
@@ -0,0 +1,432 @@
+        Args:
+            name: name of table to create
+            types: dict of column name and datatype
+
+        Returns: None
+        Raises: ValueError should table already exist
+        """
+
+    @abstractmethod
+    async def drop_table(self, name: str) -> None:
+        """
+
+        Args:
+            name: name of table to drop
+
+        Returns: None
+
+        """
+
+    @abstractmethod
+    async def insert_row(self, table_name: str, values: dict, constraint_column_name: tuple,
+                         no_conflict_resolution: bool = False,
+                         skip_double_dash_test: bool = False) -> None:
+        """
+        Args:
+            constraint_column_name: names of the unique columns.
+                These MUST be set as unique at table-creation
+            no_conflict_resolution: If there are no unique columns in your table, set this to True
+            table_name: name of table to insert value into
+            values: tuple with values matching the rows to insert into
+            skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                    implement your OWN CHECK!
+
+        Returns: None
+
+        """
+
+    @abstractmethod
+    async def update_row(self, table_name: str, connector: str, values: dict, condition=None,
+                         skip_double_dash_test=False) -> None:
+        """
+
+         Args:
+             table_name: name of table to update
+             connector: Connector used to connect conditions. Must be suported by the DB
+             values: tuple with values matching the rows to insert into
+             condition: conditions, connected by "equals"
+             skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                     implement your OWN CHECK!
+
+         Returns: None
+
+         """
+
+    @abstractmethod
+    async def delete_row(self, table_name: str, connector: str, condition=None,
+                         skip_double_dash_test: bool = False) -> None:
+        """
+
+        Args:
+            table_name: name of table to delete row form
+            connector: Connector used to connect conditions. Must be suported by the DB
+            condition: conditions, connected by "equals"
+            skip_double_dash_test: skip the crude SQLInjection test if it breaks your request,
+                    implement your OWN CHECK!
+
+        Returns: None
+
+        """
+
+
+# noinspection SqlNoDataSourceInspection
+class DatabaseManager(DBMApi, Singleton):
+    _enabled = True
+
+    # for testing only, as a makeshift "session" scope
+    marker = None
+    cursor = None
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value: bool):
+        if value == self._enabled:
+            # no need to change it if no change is required
+            return
+        if value:

```

Comment:
1. On the one hand you could argue, that passing None will also disable it (can be intended)
1. On the other hand, passing None should trigger a lethal response, as it isn't really a valid value for the request.

Reply:
1: the type hint is `bool`, not `Optional[bool]` therefor 2 is accurate. `None` should trigger a lethal response. likewise, since a daft programmer could throw anything into `value` a Truthy/Falsy check may lead to unintended regressions